src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -581,20 +581,23 @@ trait Namers extends MethodSynthesis {
         }
       }
 
-      def noDuplicates(names: List[Name], check: DuplicatesErrorKinds.Value) {
-        def loop(xs: List[Name]): Unit = xs match {
+      def noDuplicatesBy(namer: ImportSelector => Name, check: DuplicatesErrorKinds.Value): Unit = {
+        def canSkip(name: Name, sels: List[ImportSelector]): Boolean =
```

Comment:
`canSkip` seems a bit nonspecific: this function in fact asks whether `name` is nor only skippable but has received the affirmative approval of the duplicates police, right?

Reply:
So, because the condition of the `if-then-else` had grown a bit large, I cut out a local function to it. Since the `then` case is to _skip_ the head of the list, discard it as "valid" or "non-duplicated", and continue through it, well, I thought that `canSkip` was appropiate.

Note that the head's name being duplicated is only one of the reasons to skip it.

Reply:
Always a good reason to use a local function. Often a decent reason to consider `@inline`, but I agree it's not always right.

I didn't mean for this review to be all about naming things; sorry for that. A laconic `ok` is my go-to here, but I won't even "suggest" it.